### PR TITLE
Portuguese translation

### PR DESCRIPTION
**Pull request by nunohelibeires**
Reopening #4605  with clean set of commits

### CODE REVIEW

**Comment by claycrosby** [comment]
I had a native Continental Portuguese speaker review the translations and he approves. We need this for a project going live on the 1st of July.

Can we merge these translations?

**Comment by xrmx** [comment]
Unless i am missing something you haven't translated the json messages.

**Comment by mzangie** [comment]
I don't know how? I wish someone could just translate them and convert them into money for me and put them safely into a portfolio or something for me to see what I have and when it arrives?

**Comment by nunohelibeires** [comment]
@xrmx please review & warn me if something is still missing. 

@mzangie https://github.com/apache/incubator-superset/blob/master/CONTRIBUTING.md#translations
docs point to po2json, so that was what i did today.

**Comment by claycrosby** [comment]
@xrmx if you could please review and merge we would appreciate it - we have a project going live on July 1 that needs this translation

**Comment by xrmx** [comment]
@nunohelibeires @claycrosby  I don't speak portuguese and I'm not a committer :)

**Comment by benleibert** [comment]
Hey everyone - we could use this for our project as well. Any idea if and when it may be merged?

**Comment by mistercrunch** [comment]
@betodealmeida is a Portuguese-speaking committer :)

**Comment by benleibert** [comment]
Hi @betodealmeida - do you happen to have time to review this PR?

For what it's worth, @mistercrunch, I asked a native Portuguese-speaker on my team review the translations. His feedback is that the ones in this PR are quite good. Thank you very much for submitting them, @nunohelibeires. I really appreciate your work!

**Comment by nunohelibeires** [comment]
@betodealmeida do you have a spare min to review this PR ? Thanks :)

**Comment by betodealmeida** [comment]
Sorry for the delay, will review it.

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 1]
```diff
@@ -0,0 +1,4181 @@
...
+msgid "Rolling"
+msgstr "Rolling"
+
+#: superset/assets/javascripts/explore/stores/controls.jsx:718
+msgid ""
+"Defines a rolling window function to apply, works along with the [Periods] "
+"text box"
+msgstr ""
+
+#:
```
Comment (betodealmeida) [inline superset/translations/pt/LC_MESSAGES/message.po]: There's an extra space after "seleção", should be "Selecione para incluir seleção de granularidade do Druid".

Comment (betodealmeida) [inline superset/translations/pt/LC_MESSAGES/message.po]: No space after "Auto", use backtick instead of single quote.

Comment (betodealmeida) [inline superset/translations/pt/LC_MESSAGES/message.po]: "Gráfico de Queijo"?

Comment (betodealmeida) [inline superset/translations/pt/LC_MESSAGES/message.po]: "Ligação predefinida, é possível incluir {{ metric }} ou outros valores "

Comment (betodealmeida) [inline superset/translations/pt/LC_MESSAGES/message.po]: "Selecionar [%s]"

Comment (betodealmeida) [inline superset/translations/pt/LC_MESSAGES/message.po]: "Bem vindo!"

Comment (betodealmeida) [reply superset/translations/pt/LC_MESSAGES/message.po]: (Let me know if you want to add a custom logo or welcome message, I can point out how we do it in my company.)

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 2]
```diff
@@ -0,0 +1,4181 @@
...
+msgid "Mapbox"
+msgstr "Mapbox"
+
+#: superset/viz.py:1649
+msgid "Must have a [Group By] column to have 'count' as the [Label]"
+msgstr "Deve ter uma coluna [Agrupar por] para ter 'count' como [Label]"
+
+#: superset/viz.py:1662
+msgid "Choice of [Label] must be present in [Group By]"
+msgstr "A escolha da [Label] deve estar presente em []"
```
[the remark this answers is not in the document]
Comment (betodealmeida) [reply superset/translations/pt/LC_MESSAGES/message.po]: Also, should we use "Etiqueta" for "Label"?

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 3]
```diff
@@ -0,0 +1,4181 @@
...
+
+#: superset/assets/javascripts/SqlLab/components/SaveQuery.jsx:16
+msgid "Undefined"
+msgstr "Indefinido"
+
+#: superset/assets/javascripts/SqlLab/components/SaveQuery.jsx:66
+#: superset/views/sql_lab.py:53
+#, fuzzy
+msgid "Label"
+msgstr "Label"
```
Comment (betodealmeida) [inline superset/translations/pt/LC_MESSAGES/message.po]: "Etiqueta"? There are some more messages below where "Label" is translated as "Etiqueta", and some others were it's left unstranslated.

**Review by betodealmeida — CHANGES_REQUESTED** [review]
This is awesome, thank you for putting the effort in doing this translation! I have a few comments, and there are a few cases where `%s` is written as `% s`, other than that it should be good to go!

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 4]
```diff
@@ -0,0 +1,4181 @@
...
+
+#: superset/assets/javascripts/explore/stores/visTypes.js:508
+#: superset/viz.py:616
+msgid "Calendar Heatmap"
+msgstr "Calendário com Mapa de Calor"
+
+#: superset/assets/javascripts/explore/stores/visTypes.js:529
+#: superset/viz.py:674
+msgid "Box Plot"
+msgstr "Parcela"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: Yes, I agree. Box plot is the most used term.

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 5]
```diff
@@ -0,0 +1,4181 @@
...
+msgstr "Deve ser especificada uma coluna númerica"
+
+#: superset/assets/javascripts/explore/stores/visTypes.js:96
+#: superset/viz.py:1205
+msgid "Distribution - Bar Chart"
+msgstr "Distribuição - Gráfico de barras"
+
+#: superset/viz.py:1216
+msgid "Can't have overlap between Series and Breakdowns"
+msgstr "Não pode haver sobreposição entre Series e Desagregação"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: yes, thanks

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 6]
```diff
@@ -0,0 +1,4181 @@
...
+msgstr "Histograma"
+
+#: superset/viz.py:1190
+msgid "Must have one numeric column specified"
+msgstr "Deve ser especificada uma coluna númerica"
+
+#: superset/assets/javascripts/explore/stores/visTypes.js:96
+#: superset/viz.py:1205
+msgid "Distribution - Bar Chart"
+msgstr "Distribuição - Gráfico de barras"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: ok, changed

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 7]
```diff
@@ -0,0 +1,4181 @@
...
+
+#: superset/assets/javascripts/explore/stores/visTypes.js:835
+#: superset/viz.py:1409
+msgid "Country Map"
+msgstr "Mapa de País"
+
+#: superset/assets/javascripts/explore/stores/visTypes.js:868
+#: superset/viz.py:1437
+msgid "World Map"
+msgstr "Mapa Mundo"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: Although no one uses the term verbally around here, Mapa-Múndi is the correct form. Nice.

File: superset/translations/pt/LC_MESSAGES/message.po (hunk 2 above)
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: Well thinking about this one again now, I also don't like "Etiqueta" for "Label".

What about "Rótulo" @betodealmeida , I think it is used more for this situation no?

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 8]
```diff
@@ -0,0 +1,4181 @@
...
+
+#: superset/assets/javascripts/SqlLab/actions.js:184
+#, fuzzy
+msgid "Failed at stopping query."
+msgstr "Falha ao parar a query."
+
+#: superset/assets/javascripts/SqlLab/actions.js:297
+#: superset/assets/javascripts/SqlLab/actions.js:310
+msgid "Error occurred while fetching table metadata"
+msgstr "Ocorreu um erro ao carregar os metadados na tabela"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: I agree

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 9]
```diff
@@ -0,0 +1,4181 @@
...
+
+#: superset/assets/javascripts/SqlLab/components/QueryTable.jsx:49
+#: superset/assets/javascripts/SqlLab/reducers.js:11
+msgid "Untitled Query"
+msgstr "Query sem título"
+
+#: superset/assets/javascripts/SqlLab/reducers.js:44
+#, python-format
+msgid "Copy of %s"
+msgstr "Cópia de% s"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: Reviewing all of the %s

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 10]
```diff
@@ -0,0 +1,4181 @@
...
+
+#: superset/assets/javascripts/dashboard/Dashboard.jsx:358
+#: superset/assets/javascripts/dashboard/components/SaveModal.jsx:100
+msgid "Error"
+msgstr "Erro"
+
+#: superset/assets/javascripts/dashboard/Dashboard.jsx:359
+#, python-format
+msgid "Sorry, there was an error adding slices to this dashboard: %s"
+msgstr "Desculpe, houve um erro ao adicionar fatias a este painel:% s"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: "Visualização" is the best translation indeed.

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 11]
```diff
@@ -0,0 +1,4181 @@
...
+msgstr "deve ser um número inteiro"
+
+#: superset/assets/javascripts/explore/validators.js:30
+msgid "cannot be empty"
+msgstr "não pode estar vazio"
+
+#: superset/assets/javascripts/explore/components/ChartContainer.jsx:180
+#, python-format
+msgid "%s - untitled"
+msgstr "% s - sem título"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: I missed a lot of these

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 12]
```diff
@@ -0,0 +1,4181 @@
...
+msgid "Please enter a slice name"
+msgstr "Por favor insira um nome para a visualização"
+
+#: superset/assets/javascripts/explore/components/SaveModal.jsx:89
+msgid "Please select a dashboard"
+msgstr "Por favor selecione um dashboard"
+
+#: superset/assets/javascripts/explore/components/SaveModal.jsx:97
+msgid "Please enter a dashboard name"
+msgstr "Por favor insira"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: Another miss, thanks.

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 13]
```diff
@@ -0,0 +1,4181 @@
...
+msgid "Save as"
+msgstr "Gravar como"
+
+#: superset/assets/javascripts/explore/components/SaveModal.jsx:168
+msgid "[slice name]"
+msgstr "[nome da visualização]"
+
+#: superset/assets/javascripts/explore/components/SaveModal.jsx:181
+msgid "Do not add to a dashboard"
+msgstr "Não adicione um dashboard"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: Corrected.

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 14]
```diff
@@ -0,0 +1,4181 @@
...
+
+#: superset/assets/javascripts/explore/reducers/chartReducer.js:32
+msgid "Updating chart was stopped"
+msgstr "A atualização do gráfico parou"
+
+#: superset/assets/javascripts/explore/reducers/chartReducer.js:38
+#: superset/assets/javascripts/modules/superset.js:226
+#, python-format
+msgid "An error occurred while rendering the visualization: %s"
+msgstr "Ocorreu um erro ao renderizar a visualização:% s"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: In the end I will indeed, thanks.

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 15]
```diff
@@ -0,0 +1,4181 @@
...
+#: superset/assets/javascripts/explore/reducers/chartReducer.js:47
+msgid ""
+"Perhaps your data has grown, your database is under unusual load, or you "
+"are simply querying a data source that is to large to be processed within "
+"the timeout range. If that is the case, we recommend that you summarize "
+"your data further."
+msgstr ""
+"Talvez a quantidade de dados tenha aumentado, a base de dados está em "
+"sobrecarga anormal, ou está simplesmente a consultar uma origem de dados  "
+"grande de mais para ser processada dentro do intervalo de tempo limite. Se "
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: This one got me thinking, a lot. In the end, after asking a teacher of portuguese I found out that both of them are accepted and it is an old linguistic fight.
In the end I'll go with your suggestion as it implies better the meaning of quantity as it happens in this case. This was a fun one 👍

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 16]
```diff
@@ -0,0 +1,4181 @@
...
+msgid "Percentage Metrics"
+msgstr "Métricas percentuais"
+
+#: superset/assets/javascripts/explore/stores/controls.jsx:113
+msgid "Metrics for which percentage of total are to be displayed"
+msgstr "Métricas para qual porcentagem do total deve ser exibida"
+
+#: superset/assets/javascripts/explore/stores/controls.jsx:118
+msgid "Y Axis Bounds"
+msgstr "Y Axis Bounds"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: Yes, agreed.

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 17]
```diff
@@ -0,0 +1,4181 @@
...
+msgid "Choose a metric for right axis"
+msgstr "Escolha uma métrica para o eixo direito"
+
+#: superset/assets/javascripts/explore/stores/controls.jsx:187
+msgid "Stacked Style"
+msgstr "Estilo empilhado"
+
+#: superset/assets/javascripts/explore/stores/controls.jsx:199
+msgid "Sort X Axis"
+msgstr "Classificar X Axis"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: Agreed.

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 18]
```diff
@@ -0,0 +1,4181 @@
...
+
+#: superset/assets/javascripts/explore/stores/controls.jsx:207
+msgid "Sort Y Axis"
+msgstr "Eixo de classificação Y"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: Agreed.

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 19]
```diff
@@ -0,0 +1,4181 @@
...
+
+#: superset/assets/javascripts/explore/stores/controls.jsx:215
+msgid "Linear Color Scheme"
+msgstr "Esquema de cores lineares"
+
+#: superset/assets/javascripts/explore/stores/controls.jsx:233
+msgid "Normalize Across"
+msgstr "Normalize across"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: I think "Normalize em função de" works well.

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 20]
```diff
@@ -0,0 +1,4181 @@
...
+msgid "Origin"
+msgstr "Origem"
+
+#: superset/assets/javascripts/explore/stores/controls.jsx:473
+msgid ""
+"Defines the origin where time buckets start, accepts natural dates as in "
+"`now`, `sunday` or `1970-01-01`"
+msgstr ""
+"Define a origem onde os campos de tempo começam, aceita datas naturais como "
+"`agora`,`domingo` ou `1970-01-01`"
```
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: ok, makes sense

File: superset/translations/pt/LC_MESSAGES/message.po (hunk 1 above)
[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: corrected to "Métrica atribuída ao eixo [X]"

[the remark this answers is not in the document]
Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: and this one to "Metrica atribuída ao eixo [Y]"

Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: agreed

Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: changed

Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: corrected

Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: corrected

Comment (nunohelibeires) [reply superset/translations/pt/LC_MESSAGES/message.po]: changed. Well, I changed the config.py for app name & icon. If any other customization is possible yes, please let me know ;)

**Comment by nunohelibeires** [comment]
Thank you for the review @betodealmeida , I commented all your reviews and suggestions, changed the .po file and uploaded the correspondent .mo file.

File: superset/translations/pt/LC_MESSAGES/message.po [hunk 21]
```diff
@@ -0,0 +1,4181 @@
...
+msgstr "SQL"
+
+#: superset/assets/javascripts/SqlLab/components/QueryHistory.jsx:28
+msgid "No query history yet..."
+msgstr "Ainda não há histórico de queries ..."
+
+#: superset/assets/javascripts/SqlLab/components/QuerySearch.jsx:106
+#: superset/assets/javascripts/SqlLab/components/SqlEditorLeftBar.jsx:66
+msgid "It seems you don't have access to any database"
+msgstr "Parece que não tem acesso a nenhuma base de dados"
```
[the remark this answers is not in the document]
Comment (betodealmeida) [reply superset/translations/pt/LC_MESSAGES/message.po]: Sounds good to me! I know there's a lot of differences between pt and pt_BR.

**Review by betodealmeida — COMMENTED** [review]
Awesome!

**Review by betodealmeida — APPROVED** [review]

**Comment by csc5k** [comment]
Hi everyone, thanks for getting this together. When can we expect it to be incorporated into the next update, and when can we expect to be able to make that update?

**Comment by nunohelibeires** [comment]
@mistercrunch Everything reviewed thanks to @betodealmeida , I think this one is ready.
Do you think it can be merged and ready for the next vers?

**Comment by csc5k** [comment]
Hi team, I see this was merged to master a few weeks ago but I'm not seeing it in the recent v27 update. Is it coming out in v28?

**Comment by nunohelibeires** [comment]
@csc5k indeed it is only in the v0.28 branch.
probably this PR was merged only after the creation of the v0.27 branch.

**Pull request merged**